actors/runtime/src/runtime/fvm.rs

```diff
@@ -120,20 +121,21 @@ where
 
     fn validate_immediate_caller_type<'a, I>(&mut self, types: I) -> Result<(), ActorError>
     where
-        I: IntoIterator<Item = &'a Cid>,
+        I: IntoIterator<Item = &'a Type>,
     {
         self.assert_not_validated()?;
 
-        let caller_addr = self.message().caller();
-        let caller_cid = self
-            .get_actor_code_cid(&caller_addr)
-            .expect("failed to lookup caller code");
-        if types.into_iter().any(|c| *c == caller_cid) {
-            Ok(())
-        } else {
-            return Err(actor_error!(SysErrForbidden;
+        let caller_cid = {
+            let caller_addr = self.message().caller();
+            self.get_actor_code_cid(&caller_addr)
+                .expect("failed to lookup caller code")
+        };
+
+        match self.is_builtin_actor(&caller_cid) {
+            Some(typ) if types.into_iter().any(|t| *t == typ) => Ok(()),
```

Comment:
Could use a bimap here to avoid the reverse lookup, but not sure it's worth it. Will do if it shows up in benchmarks.